spec/concurrent/promise_spec.rb

```diff
@@ -368,6 +368,27 @@ def get_ivar_from_args(opts)
 
         expect(composite).to be_rejected
       end
+
+      it 'preserves ordering of the executed promises' do
+        promise1 = Promise.execute do
+          # resolves after the second promise
+          sleep 0.2
```

Comment:
You shouldn't rely on `sleep` for thread synchronization/ordering. Instead use concurrency primites:

```ruby
running = Mutex.new
cond = ConditionVariable.new
cond2 = ConditionVariable.new

p1 = Concurrent::Promise.execute do
  running.synchronize {
    cond.wait(running)
    'one'
  }
end

p2 = Concurrent::Promise.execute do
  running.synchronize {
    cond2.wait(running)
    'two'
  }
end

p3 = Concurrent::Promise.execute do
  running.synchronize {
    'three'
  }
end

cond2.signal
cond.signal

result = Concurrent::Promise.zip(p1, p2, p3).value
expect(result) .to eq(['one', 'two', 'three'])
```